app/controllers/AdditionalPaymentAmountController.scala

```diff
@@ -39,16 +41,16 @@ class AdditionalPaymentAmountController @Inject()(
   sessionRepository: SessionRepository,
   val navigator: Navigator,
   identify: IdentifierAction,
-  feature: FeatureFlagActionProvider,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
   formProvider: AdditionalPaymentAmountFormProvider,
   val controllerComponents: MessagesControllerComponents,
-  view: AdditionalPaymentAmountView
-)(implicit ec: ExecutionContext)
+  view: AdditionalPaymentAmountView,
+)(implicit ec: ExecutionContext, appConf: FrontendAppConfig, eh: ErrorHandler)
     extends BaseController {
 
   val form: Form[Amount] = formProvider()
+  val feature: FeatureFlagActionProvider = new FeatureFlagActionProviderImpl()
```

Comment:
do we need separate instance in every controller where ever we this ?, i think we can inject a singleton ?

Reply:
and i see you are not using the `feature` val or did i miss anything

Reply:
it used on line 55

Reply:
ah ok, no worries, why do we need to create it using `new` instead of injecting as before - is it causing any issue with the tests ?

Reply:
there is a bit to talk about that....for now let's just say that the current code wasn't written in TDD or at least not driven by unit test - it was very tricky to stub and/or override anything. this was the easiest and probably the only way